akka-actor/src/main/scala/akka/actor/ActorSystemSettings.scala

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.actor
+
+import java.util.Optional
+
+import scala.annotation.varargs
+import scala.compat.java8.OptionConverters._
+import scala.reflect.ClassTag
+
+/**
+ * Marker trait for a setting that can be put inside [[ActorSystemSettings]]
+ */
+trait ActorSystemSetting
+
+object ActorSystemSettings {
+
+  val empty = new ActorSystemSettings(Map.empty)
+
+  /**
+   * Scala API:
+   */
+  def apply(settings: ActorSystemSetting*): ActorSystemSettings =
```

Comment:
I don't like the idea of having it mandatory because I think that the whole point of these programatic settings is that it will eliminate the need to pass a config, so then it will require a chunk of boilerplate repeated everywhere (kind of countering the idea of having optional settings):
```
ActorSystemSettings(
  ActorSystemBootstrapSettings(), 
  RemoteSettings(port = 0, host = "localhost"))
```

I'll try with the `ActorSystem(name, bootstrap)` and add some more docs to aid discoverability.